horace_core/sqw/@sqw/sqw_eval.m

```diff
@@ -25,41 +25,36 @@
 %                   c1,c2,...   Other constant parameters e.g. file name for look-up
 %                              table
 %
-%   pars        Arguments needed by the function. Most commonly, a vector of parameter
+%   pars       Arguments needed by the function. Most commonly, a vector of parameter
 %              values e.g. [A,js,gam] as intensity, exchange, lifetime. If a more general
 %              set of parameters is required by the function, then
 %              package these into a cell array and pass that as pars. In the example
 %              above then pars = {p, c1, c2, ...}
 %
-%   'all'       [option] Requests that the calculated sqw be returned over
+% Optional parameter:
+%   'all'      Requests that the calculated sqw be returned over
 %              the whole of the domain of the input dataset. If not given, then
 %              the function will be returned only at those points of the dataset
 %              that contain data.
 %               Applies only to input with no pixel information - it is ignored if
 %              full sqw object.
 %
-%   'ave'       [option] Requests that the calculated sqw be computed for the
+%   'ave'      Requests that the calculated sqw be computed for the
 %              average values of h,k,l of the pixels in a bin, not for each
 %              pixel individually. Reduces cost of expensive calculations.
-%               Applies only to the case of sqw object with pixel information - it is
-%              ignored if dnd type object.
+%              Applies only to the case of sqw object with pixel information 
+%             - it is ignored if dnd type object.
 %
 % Output:
 % -------
 %   wout        Output dataset or array of datasets
 
 
 % Check optional argument
-all_bins=false;
-ave_pix=false;
-if exist('opt','var')  % no option given
-    if ischar(opt) && ~isempty(strmatch(lower(opt),'all'))    % option 'all' given
-        all_bins=true;
-    elseif ischar(opt) && ~isempty(strmatch(lower(opt),'ave'))    % option 'ave' given
-        ave_pix=true;
-    else
-        error('Unrecognised option')
-    end
+options = {'all','average'};
```

Comment:
'average' doesn't match the optional argument given in the header ('ave')

Reply:
standard parser truncates input options to minimum difference e.g. you can enter only `al` and `av`

Reply:
or anything up to the full word

Reply:
should the option be named fully in the header comment then, so a user knows they can use 'average'?

Reply:
probably correct. Just Toby took ave which will still work. Changed.

Reply:
done